API.md

```diff
@@ -0,0 +1,58 @@
+# SegM8Pi API
+
+## class SegM8
+
+Create an object of type `SegM8` to communicate with a chain of a particular [SegM8 7-segment
+indicator modules](https://my.amperka.com/modules/SegM8).
+
+### `SegM8(pin_CE: int, device_count: int = 1)`
+
+Construct a new SegM8 object that uses the default hardware SPI bus.
+
+- `pin_CE`: the chip enable (also known as chip select or slave select) pin used to control the
+  shift-register latch. It can take the values **0** or **1**, which corresponds to the **CE0** or
+  **CE1** pins on the Raspberry Pi board.
+- `device_count`: the number of SegM8 modules connected in a daisy-chain. If omitted, defines
+  a single module.
+
+### `clear() -> None`
+
+Sets all module segments to the "dark" state. Clear the internal buffer.
+
+### `display(number: int, position: int, width: int, flags: int = SEGM8_ALIGN_RIGHT) -> None`
+
+### `display(number: float, position: int, width: int, precision: int = 1, flags: int = SEGM8_ALIGN_LEFT) -> None`
+
+### `display(string: str, position: int, width: int, flags: int = SEGM8_ALIGN_LEFT) -> None`
+
+Prints a number or a text line of a fixed `width` starting with the `position`to the output buffer
+using formatting `flags`. Moves the contents of the buffer to the indicator daisy-chain.
+Note: the `.` sign does not occupy a separate module and is displayed with the previous character.
+
+When printing text strings, the interpretation of characters in indicator segments is used. Instead
+of unsupported characters (e.g. **K, M, V, W, X, Z**) an overline is displayed (segment a).
+
+- `number`: integer or floating point numbers.
+- `string`: text line.
+- `position`: starting position of the output of a string or number in the internal buffer.
+- `width`: the number of elements in the internal buffer, and accordingly the number of SegM8
+  modules needed to display a number or a text.
+- `precision`: decimal places count of float.
+- `flag`: formatting flags. You can add somewhat consistent flags separated by `|` sign.
+
+Available formatting flags:
+
+- `SEGM8_ALIGN_RIGHT` – align to the right corner. Default state.
+- `SEGM8_ALIGN_LEFT` – align to the left corner.
+- `SEGM8_PAD_ZEROS` – add leading zeros before the number. Compatible only with `SEGM8_ALIGN_RIGHT`.
+- `SEGM8_RADIX_10` – use the decimal numeral system. Default state.
```

Comment:
Idea: 
Python has no problems with allocating dynamic arrays, and we can make the radix not only 10 or 16, but arbitrary. Users will be able to display numbers based on base 2, 8 ... or other meaningful ones. This is very simple in the program.

Reply:
Can you remember the last time you need to show an octal number on a display? :)

Reply:
Not so long ago. While experimenting with msp430. There assembler is a direct successor of PDP11.

Reply:
I also think this is overkill for this module.

Reply:
Redeployment is removed :)